engine/engine_test.go

```diff
@@ -309,6 +309,32 @@ func TestQueriesAgainstOldEngine(t *testing.T) {
 			end:   time.Unix(3230, 0),
 			step:  28 * time.Second,
 		},
+		{
+			name: "topk",
+			load: `load 30s
+				http_requests_total{pod="nginx-1", series="1"} 1+1.1x40
+				http_requests_total{pod="nginx-2", series="2"} 2+2.3x50
+				http_requests_total{pod="nginx-4", series="3"} 5+2.4x50
+				http_requests_total{pod="nginx-5", series="1"} 8.4+2.3x50
+				http_requests_total{pod="nginx-6", series="2"} 2.3+2.3x50`,
```

Comment:
For this, maybe having test cases with series that are not monotonically increasing would help better in catching any inaccuracies?